### DISCUSSION TITLE
how to use with python

### DISCUSSION THREAD

**Discussion by famko09** [LibreHardwareMonitor/LibreHardwareMonitor#677] [Q&A]
I am using a python script  that's get the sensors' data from  OpenHardwareMonitorLib.dll. Is this also possible with  the LibreHardwareMonitorLib.dll ? 
Just renaming the OpenHardwareMonitor to LibreHardwareMonitor isn't working.

Does anyone have an idea how to get this work?

Any help will be appreciated 

here is the python example: 
``
import clr #package pythonnet, not clr

openhardwaremonitor_hwtypes = ['Mainboard','SuperIO','CPU','RAM','GpuNvidia','GpuAti','TBalancer','Heatmaster','HDD']

openhardwaremonitor_sensortypes = ['Voltage','Clock','Temperature','Load','Fan','Flow','Control','Level','Factor','Power','Data','SmallData']

def initialize_openhardwaremonitor():
    file = 'OpenHardwareMonitorLib'
    clr.AddReference(file)

    from OpenHardwareMonitor import Hardware

    handle = Hardware.Computer()
    handle.MainboardEnabled = True
    handle.CPUEnabled = True
    handle.RAMEnabled = True
    handle.GPUEnabled = True
    handle.HDDEnabled = True
    handle.Open()
    return handle

def fetch_stats(handle):
    for i in handle.Hardware:
        i.Update()
        for sensor in i.Sensors:
            parse_sensor(sensor)
        for j in i.SubHardware:
            j.Update()
            for subsensor in j.Sensors:
                parse_sensor(subsensor)

def parse_sensor(sensor):
        if sensor.Value is not None:
            
            if type(sensor).__module__ == 'OpenHardwareMonitor.Hardware':
                sensortypes = openhardwaremonitor_sensortypes
                hardwaretypes = openhardwaremonitor_hwtypes
            else:
                return

            if sensor.SensorType == sensortypes.index('Temperature'):
                print(u"%s %s Temperature Sensor #%i %s - %s\u00B0C" % (hardwaretypes[sensor.Hardware.HardwareType], sensor.Hardware.Name, sensor.Index, sensor.Name, sensor.Value))

if __name__ == "__main__":
    print("OpenHardwareMonitor:")
    HardwareHandle = initialize_openhardwaremonitor()
    fetch_stats(HardwareHandle)
``

**Comment by lxvko** [comment]
Found a solution in this repository: https://github.com/NaturalBornCamper/system-monitor

Simplified: 
`

      import os
      import clr #pythonnet
      
      
      def initialize():
      
          clr.AddReference(os.getcwd() + "\\LibreHardwareMonitorLib.dll")
      
          from LibreHardwareMonitor import Hardware
      
          handle = Hardware.Computer()
          handle.IsCpuEnabled = True
          handle.IsGpuEnabled = True
          handle.IsMemoryEnabled = True
          handle.IsMotherboardEnabled = True
          handle.IsControllerEnabled = True
          handle.IsNetworkEnabled = True
          handle.IsStorageEnabled = True
          handle.Open()
          return handle
      
      
      def fetch_stats(handle):
          for i in handle.Hardware:
              i.Update()
              for sensor in i.Sensors:
                  parse_sensor(sensor)
      
      
      def parse_sensor(sensor):
          if sensor.Value is not None:
              data = (f"{sensor.Hardware.Name} "
                           f"{sensor.Name} "
                           f"{sensor.Value}")
              print(data)


      if __name__ == "__main__":
          HardwareHandle = initialize()
          fetch_stats(HardwareHandle)
      
`

**Comment by famko09** [comment]
Yes Thanks,
had to make a minor change to get it work, don't know why, but i have changed 

clr.AddReference(os.getcwd() + "\\LibreHardwareMonitorLib.dll")

by
      file = 'LibreHardwareMonitorLib'
      clr.AddReference(file)

It'sworking, thank you
